src/modules/job-ingest/validate.c

```diff
@@ -61,11 +61,36 @@ struct validate {
     struct worker *worker[MAX_WORKER_COUNT];
 };
 
+static void validate_killall (struct validate *v)
+{
+    flux_future_t *cf = flux_future_wait_all_create ();
+    flux_future_t *f;
+    int i;
+    if (!cf) {
+        flux_log_error (v->h, "validate_destroy: flux_future_wait_all_create");
+        return;
+    }
+    flux_future_set_flux (cf, v->h);
+    for (i = 0; i < MAX_WORKER_COUNT; i++) {
+        if ((f = worker_kill (v->worker[i], SIGKILL)))
+            flux_future_push (cf, NULL, f);
+    }
+    /* Wait a bit until signal has been delivered to all workers
+     *  before continuing, hopefully ensuring workers are not left
+     *  around. (report but o/w ignore errors)
+     */
+    if (flux_future_wait_for (cf, .100) < 0
```

Comment:
What would be the consequence of waiting without a timeout here?  Is there a scenario where the broker would not respond?  It seems like the timeout increases the probability for timing related non-determinism in slow environments (although the alternate code paths are a lot safer thanks to your other recent PRs!) . If it's motivated by an abundance of caution, maybe a longer timeout would make sure it only triggers when something goes wrong?

Reply:
Yeah, it was an abundance of caution.  At first I had it at 5s, then lowered it for testing purposes, then figured it didn't matter and kept it that way. I just didn't want to introduce a fully blocking call into the module's exit path since the job-ingest module is unloaded synchronously during rc3.

On the other hand, I don't think there'd be a huge impact if the module didn't even wait for the RPC response at all in this codepath. We'd just very rarely be back where we are now -- a worker process might be left around if you reload the module in a running instance. (During instance shutdown the broker exec service will eventually kill off all subprocesses anyway.)

:man_shrugging:

Reply:
(BTW, the timeout does seem to hit occasionally during `t5000-valgrind.t` with no bad side effects .. yet)

Reply:
Ok, force-pushed the timeout back to 5s.